test/JDBC/expected/1_grant_schema.out

```diff
@@ -0,0 +1,2936 @@
+go
+revoke select on s1.v1 from public;
+go
+revoke execute on s1.p1 from public;
+go
+revoke execute on s1.f1 from public;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 11 [select, insert, delete, execute privilege on these objects to u1]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
+go
+~~START~~
+int
+~~END~~
+
+insert into s1.t1 values (1);
+go
+~~START~~
+varchar
+insert successful
+~~END~~
+
+~~ROW COUNT: 1~~
+
+delete from s1.t1 where a = 1;
+go
+~~ROW COUNT: 1~~
+
+select * from s1.v1;
+go
+~~START~~
+int
+1
+~~END~~
+
+exec s1.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from s1.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+delete from s1.t1 where a = 1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql
+grant select, insert, delete, execute on schema::s1 to public;
+go
+grant select, insert, delete, execute on schema::s1 to u1;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 12
+-- [select, insert, delete, execute privilege on these objects to u1]
+-- [select, insert, delete, execute privilege on schema to public]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
+go
+~~START~~
+int
+~~END~~
+
+insert into s1.t1 values (1);
+go
+~~START~~
+varchar
+insert successful
+~~END~~
+
+~~ROW COUNT: 1~~
+
+delete from s1.t1 where a = 1;
+go
+~~ROW COUNT: 1~~
+
+select * from s1.v1;
+go
+~~START~~
+int
+1
+~~END~~
+
+exec s1.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from s1.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t1;
+go
+~~START~~
+int
+~~END~~
+
+insert into s1.t1 values (1);
+go
+~~START~~
+varchar
+insert successful
+~~END~~
+
+~~ROW COUNT: 1~~
+
+delete from s1.t1 where a = 1;
+go
+~~ROW COUNT: 1~~
+
+select * from s1.v1;
+go
+~~START~~
+int
+1
+~~END~~
+
+exec s1.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from s1.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql
+revoke select, insert, delete, execute on schema::s1 from u1;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 13
+-- [select, insert, delete, execute privilege on these objects to u1]
+-- [select, insert, delete, execute privilege on schema to public, u1]
```

Comment:
select, insert, delete, execute privilege on schema to public.

We have revoked on schema s1 from u1 above